flink-runtime/src/main/java/org/apache/flink/runtime/leaderelection/LeaderElectionUtils.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.leaderelection;
+
+import org.apache.flink.util.Preconditions;
+
+import org.apache.flink.shaded.guava30.com.google.common.base.Joiner;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class LeaderElectionUtils {
+
+    public static String convertToString(
+            Map<String, LeaderInformation> leaderInformationPerContender) {
+        final Map<String, String> convertedMap =
```

Comment:
Having to construct another map purely for value string conversions seems wrong.
```
map.entrySet().stream()
  .map(entry -> entry.getKey() + "=" + convertToString(entry.getValue())
  .collect(Collectors.joining(", "))
```

Reply:
I'm gonna remove these methods for now. They are not used yet but will be used in a future PR.